### ISSUE TITLE
https://www.google.com/recaptcha/api2/pat 401 (Unauthorized)

### ISSUE THREAD

**Issue by arebelof** [google/recaptcha#561]
Hi,

When I have the Chrome's Device Toolbar active (CTRL + SHIFT + M), I'm getting this error on console when loading the page:

recaptcha__pt.js:805 
  **POST https://www.google.com/recaptcha/api2/pat?k=<My Key> 401 (Unauthorized)**

If I disable the device tollbar, there's no error.

![image](https://github.com/user-attachments/assets/1ead229d-b1e6-46d3-9a64-18e2961e1d4a)

![image](https://github.com/user-attachments/assets/136e72f1-901b-4651-876b-0f676991d9f6)

![image](https://github.com/user-attachments/assets/2e6a9cc8-2cbe-4b03-b0ef-7e3851d98274)

Thank you.

**Comment by mchamma** [comment]
Same

**Comment by guncv** [comment]
same here how to solve it

**Comment by Berman59** [comment]
Same!

**Comment by notKamui** [comment]
Please stop saying "Same" like parrots. You're polluting the issue thread. Instead, upvote the initial comment, and subscribe to the issue

![Image](https://github.com/user-attachments/assets/2231d548-c8ed-4f15-89f6-5e2913c9f66a)

**Comment by cududa** [comment]
Additionally, I've noticed that when I use Safari dev tools to remotely connect to my iPhone and pull up dev tools in Safari, it seems this error occurs on every load

**Comment by jeremy-hunter** [comment]
cududa - thank you for sharing, that totally cleared everything up for me

Overriding the user agent without enabling device mode will still exhibit the same behavior

This user agent value does NOT exhibit this behavior in chrome's dev tools
`Mozilla/5.0 (iPhone; CPU iPhone OS 13_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/129.0.0.0 Mobile/15E148 Safari/604.1`
![Image](https://github.com/user-attachments/assets/91485b48-c302-4423-a6b5-43a6563b5fe4)

This user agent value DOES exhibit this behavior in chrome's dev tools
`Mozilla/5.0 (iPhone; CPU iPhone OS 16_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Mobile/15E148 Safari/604.1`
![Image](https://github.com/user-attachments/assets/6c6c87d4-4282-4349-873e-05245dcb91c4)

**Comment by cududa** [comment]
@jeremy-hunter 

Yes, different user agents change the in dev tools - but in prod, testing on a real device I still get a 401.

I'm hoping Google doesn't close this issue, as it appears recaptcha just won't load on recent iOS versions and there's no guidance or documentation on the issue.

**Comment by NareshRoka** [comment]
Seems like the problem start when user agent has  iPhone OS 16_0 and above. Its working fine till version  iPhone OS 15_8.

**Comment by dbrisson-passat** [comment]
anyone has solved this issue please ?

**Comment by Rick-EESCOC** [comment]
I see this issue on Safari Version 17.5 (19618.2.12.11.6)
Is this because of CORS? 
![Image](https://github.com/user-attachments/assets/de689243-2042-4ba4-b73c-d556df94a7f4)

Working on Chrome  130.0.6723.92 (Official Build) (x86_64)

Working on Chrome Canary 132.0.6817.0 

Working on Firefox latest build. 132.0 

Not checking other Chromium browsers.

**Comment by kemalkastrataj** [comment]
Alright, guys, I had the same issue, and the problem was with how my key was being encoded.

I fixed it by updating this:
```
data =
  data +
  "&action=" +
  actionName +
  "&recaptcha_response=" +
  token;
```
To this:
```
data =
  data +
  "&action=" +
  encodeURIComponent(actionName) +
  "&recaptcha_response=" +
  encodeURIComponent(token);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

**Comment by nstiac** [comment]
Same here .. confirmed it happens only with iphone user agent .. but still happening.

**Comment by kemalkastrataj** [comment]
> Same here .. confirmed it happens only with iphone user agent .. but still happening.

I can confirm this; it is the same issue on my sites.

**Comment by dbrisson-passat** [comment]
anyone have a solution for this issue or no?

**Comment by jasminmistry** [comment]
Is there any temporary solution for this?

**Comment by geanfarias** [comment]
there is some update?

**Comment by elielson-anjos** [comment]
Peas as always breaking everything and the developers has to fix theyr s*****. Any fix on this? This is a super bad bug...

**Comment by tommag21** [comment]
I wouldn't expect an answer in this repostory. I opened a discussion in the Google Cloud Security forums If anyone wants to contribute: https://www.googlecloudcommunity.com/gc/reCAPTCHA/reCAPTCHA-initialization-error-on-Safari-iOS-16/td-p/859376

**Comment by mmfarahat** [comment]
is there are updates?

**Comment by mericlomlu** [comment]
I do not know if this helps but I found this thread:

https://www.googlecloudcommunity.com/gc/reCAPTCHA/reCAPTCHA-initialization-error-on-Safari-iOS-16/td-p/859376

**Comment by notKamui** [comment]
So TL;DR

Google are ducking clowns, and you can "safely ignore" the error, and they don't even plan on fixing it.

This is absolutely insane.

**Comment by fabswt** [comment]
Wasted too much time on this.

TL;DR: in the Chrome DevTools for iOS devices (iPhone, iPad...) we get a 401 on this URL:

```
https://www.google.com/recaptcha/api2/pat?k=‹google recaptcha client key›
```

It's annoying and polluting the console log. Does not prevent recaptcha from working. Does not occur on non-iOS in my experience.

**Comment by r4g3ch33m5** [comment]
same

**Comment by chris1797** [comment]
I ran into this issue while developing the SMS authentication part of FireBase. Is this by any chance a configuration issue in the FireBase console?

**Comment by agustinFeijoo** [comment]
Same

**Comment by almera-vs** [comment]
same issue, safari 18.3.1 (20620.2.4.11.6)

**Comment by sebastianfeistl** [comment]
For anyone still running into this:
This error seems to be related to Apple’s Private Access Tokens (PAT) feature on iOS/macOS devices, triggering a 401 on `api2/pat`.

Google staff confirmed [here](https://www.googlecloudcommunity.com/gc/reCAPTCHA/reCAPTCHA-initialization-error-on-Safari-iOS-16/td-p/859376) that it’s safe to ignore, as it doesn’t affect reCAPTCHA functionality. They're probably not gonna fix it.

Just one of those things—ignore it and carry on.
